api/src/main/scala/za/co/absa/loginsvc/rest/service/JWTService.scala

```diff
@@ -76,8 +76,11 @@ class JWTService @Autowired()(jwtConfigProvider: JwtConfigProvider) {
       .setIssuedAt(issuedAt)
       .claim("kid", publicKeyThumbprint)
       .claim("groups", groupsClaim)
-      .applyIfDefined(user.email)((builder, value: String) => builder.claim("email", value))
-      .applyIfDefined(user.displayName)((builder, value: String) => builder.claim("displayname", value))
+      .addClaims(
+        user.optionalAttributes.collect {
+          case (key, Some(value)) => key -> value
+        }.asJava
+      )
```

Comment:
great, this part is key - there should be no change to the current expected fields 👍